src/vaults/SlippageVault.sol

```diff
@@ -0,0 +1,231 @@
+            uint232 balanceLastHarvest = strategies[strategy].balance;
+            uint256 balanceThisHarvest = strategy.totalLockedValue();
+
+            // Update the strategy's stored balance.
+            strategies[strategy].balance = uint232(balanceThisHarvest);
+
+            // Increase/decrease newTotalStrategyHoldings based on the profit/loss registered.
+            // We cannot wrap the subtraction in parenthesis as it would underflow if the strategy had a loss.
+            newTotalStrategyHoldings = newTotalStrategyHoldings + balanceThisHarvest - balanceLastHarvest;
+
+            unchecked {
+                // Update the total profit accrued while counting losses as zero profit.
+                // Cannot overflow as we already increased total holdings without reverting.
+                totalProfitAccrued += balanceThisHarvest > balanceLastHarvest
+                    ? balanceThisHarvest - balanceLastHarvest // Profits since last harvest.
+                    : 0; // If the strategy registered a net loss we don't have any new profit.
+            }
+        }
+
+        if (totalProfitAccrued > 0) {
+            uint256 perfFee = totalProfitAccrued.mulDiv(performanceFeeBps, MAX_BPS, MathUpgradeable.Rounding.Up);
+            accumulatedPerformanceFee += uint128(perfFee);
+            totalProfitAccrued -= perfFee;
+            newTotalStrategyHoldings -= perfFee;
+            // Update max unlocked profit based on any remaining locked profit plus new profit.
+            maxLockedProfit = uint128(lockedProfit() + totalProfitAccrued);
+        } else if (newTotalStrategyHoldings < oldTotalStrategyHoldings) {
+            // updating locked profit with loss
+            if (lockedProfit() > (oldTotalStrategyHoldings - newTotalStrategyHoldings)) {
+                // Update max unlocked profit removing the loss
+                maxLockedProfit = uint128(lockedProfit() - (oldTotalStrategyHoldings - newTotalStrategyHoldings));
+            }
+        }
+
+        // Set strategy holdings to our new total.
+        totalStrategyHoldings = newTotalStrategyHoldings;
+
+        // Assess fees (using old lastHarvest) and update the last harvest timestamp.
+        _assessFees();
+        lastHarvest = uint128(block.timestamp);
+    }
+
+    function harvest(Strategy[] calldata strategyList) external virtual override onlyRole(HARVESTER) {
+        // Profit must not be unlocking
+        // require(block.timestamp >= lastHarvest + LOCK_INTERVAL, "BV: profit unlocking");
+
+        // Get the Vault's current total strategy holdings.
+        uint256 oldTotalStrategyHoldings = totalStrategyHoldings;
+
+        // Used to store the new total strategy holdings after harvesting.
+        uint256 newTotalStrategyHoldings = oldTotalStrategyHoldings;
+
+        // Used to store the total profit accrued by the strategies.
+        uint256 totalProfitAccrued;
+
+        for (uint256 i = 0; i < strategyList.length; i = uncheckedInc(i)) {
```

Comment:
It'd be nice if we could pull the logic for harvesting a single strategy into its own function.